src/odb/src/lefin/lefMacroClassParser.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the University nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <iostream>
+#include <string>
+
+#include "boostParser.h"
+#include "db.h"
+#include "lefMacroPropParser.h"
+#include "lefin.h"
+
+namespace lefMacroClassType {
+
+template <typename Iterator>
+bool parse(Iterator first, Iterator last, odb::dbMaster* master) {
+  qi::rule<std::string::iterator, space_type> TypeRule =
+      (lit("CLASS") >>
+       (lit("ENDCAP") >>
+        (lit("BOTTOMEDGE")[boost::bind(
+             &odb::dbMaster::setType, master,
```

Comment:
When I started on this I did:

  auto make_lit = [master](const char* name, dbMasterType::Value type) {
    return lit(name)[boost::bind(&odb::dbMaster::setType, master, type)];
  };

to save on save on repetition